### PR TITLE
Adds metric-based testing

### LINKED ISSUE #64: Metric-based unit tests
# Current
Currently, our unit tests for the main model training code ultimately rely on diffs between the files. 

# Objective
On top of (not replacing) the diff tests, we should also add tests to check metrics functionality

### PR DESCRIPTION
**Pull request by kylebgorman**
This PR first improves the prediction-based testing by very lightly tuning the two models:

* We now use language-appropriate pretrained models
* We increase the learning rates substantially; the goal is to overfit here anyways
* We now do 10 epochs of training; this is not very expensive compared to the time spent downloading dependencies and pretrained models
* We actually reuse the best checkpoint (!)

This gives us, to rounding, near-perfect resubstitution accuracy, so any model which gives the expected predictions has successfully overfit. To confirm this, we run `udtube test` and confirm that we get the same results.

This is an lightweight solution and closes #64.

Other details:

* I moved some testdata files around.
* I changed `assertFileExists` to `assertNonEmptyFileExists`, just to be sure. `os.stat` is very cheap.
* I added docs on how to regenerate the data if models ever change; this should make fixing this change-detector test much easier!

### CODE REVIEW

**Comment by kylebgorman** [comment]
@DanielYakubov just improved this a bit if you've already looked...just FYI

**Review by DanielYakubov — APPROVED** [review]
LGTM, no comments
